src/main/java/org/fogbowcloud/app/core/dto/TaskSpecDTO.java

```diff
@@ -1,44 +1,51 @@
 package org.fogbowcloud.app.core.dto;
 
-import org.fogbowcloud.app.core.command.Command;
 import org.fogbowcloud.app.core.task.Specification;
-
 import java.io.Serializable;
 import java.util.List;
 import java.util.Map;
 
 public class TaskSpecDTO implements Serializable {
+    /**
+     * 
+     */
+    private static final long serialVersionUID = 1L;
+    
+    private String id;
     private Specification spec;
     private List<String> commands;
     private Map<String, String> metadata;
 
-    public TaskSpecDTO(Specification spec, List<String> commands, Map<String, String> metadata){
+    public TaskSpecDTO(String id, Specification spec, List<String> commands,
+            Map<String, String> metadata) {
+        this.id = id;
         this.spec = spec;
         this.commands = commands;
         this.metadata = metadata;
     }
 
-    public TaskSpecDTO() { }
+    public TaskSpecDTO() {}
```

Comment:
do we need this empty constructor due to the gson, right? drop a comment to remember us it cannot be removed